### ISSUE TITLE
Desenvolvimento - Semana 17 de Maio

### ISSUE THREAD

**Issue by alealcoforado** [alealcoforado/zeroberto#3]
- [x] 1) Implementar proibição de reposição (até a ultima etapa): @thomas-ferraz 
- [x] 2) Implementar alternância de métodos de seleção durante o treinamento: @thomas-ferraz 
- [x] 3) Embeddings são atualizados a cada iteração. Mudar para usar os originais sempre (trainer pega do 1o modelo e armazena) (definir se os embeddings proibidos vao na clusterizaçao ou nem isso): @thomas-ferraz 
- [x] 4) Expor as learning rates como hiperparametros: Alê
- [x] 5) Implementar body_epochs e head_epochs (sobrescreve num_epochs se differentiable_head = True): @alealcoforado 
- [x] 6) Implementar temperatura (dividir no softmax): @thomas-ferraz 

- [x] x) Embeddings indo para CPU (analisar): Thomas
- [x] x) Implementar Last-Shot (reseta modelo no final e faz um treinão): TBD

**Comment by thomas-ferraz** [comment]
For itens 2 and 6 we should add this to the train_eval.py to generate the expected effect - @alealcoforado
